sortinghat/core/api.py

```diff
@@ -888,3 +888,104 @@ def _delete_unique_identities(trxl, uids):
+    identities are modified.
+
+    When a given identity `uuid` is equal to the `uuid` from its parent unique
+    identity, there will be no effect.
+
+    The function raises a `NotFoundError` exception when either any `uuid` from
+    the list does not exist in the registry.
+
+    The function raises an `InvalidValueError` exception when either any `uuid`
+    from the list is `None` or an empty string.
+
+    :param ctx: context from where this method is called
+    :param uuids: List of identifiers of the identities set to be unmerged
+
+    :returns: a list of UniqueIdentity objects as the result of the unmerged identities
+
+    :raises NotFoundError: raised when when either any `uuid` from the list
+        does not exist in the registry
+    :raises InvalidValueError: raised when when either any `uuid` from the
+        list is `None` or an empty string
+    """
+
+    def _find_identities(uuids):
+        """Find the identities to be unmerged from their parent unique identities"""
+
+        identities = []
+        for uuid in uuids:
+            # Check whether input values are valid
+            if uuid is None:
+                raise InvalidValueError(msg="'uuid' cannot be None")
+            if uuid == '':
+                raise InvalidValueError(msg="'uuid' cannot be an empty string")
+
+            try:
+                uid = find_identity(uuid)
+                identities.append(uid)
+            except NotFoundError as exc:
+                raise exc
+
+        return identities
+
+    def _set_destination_identity(trxl, identity):
+        """
+        Create the new unique identity where the identity
+        will be moved to if it does not exist
+        """
+        if identity.id != identity.uidentity_id:
+            uid = add_unique_identity_db(trxl, identity.id)
+            uid = update_profile_db(trxl, uid,
+                                    name=identity.name,
+                                    email=identity.email)
+        else:
+            # Case when the identity to be unmerged is the main one
+            uid = identity.uidentity
+
+        return uid
+
+    def _move_to_destination(trxl, identity, uid):
+        """Move the identity from the old unique identity to the new one"""
+        try:
+            uidentity = move_identity_db(trxl, identity, uid)
+        except ValueError:
+            # Case when the identity is already assigned to the unique identity
+            uidentity = uid
+
+        return uidentity
+
+    # Check input value
+    if uuids is None:
+        raise InvalidValueError(msg="'uuids' cannot be None")
+    if uuids == []:
+        raise InvalidValueError(msg="'uuids' cannot be an empty list")
+
+    trxl = TransactionsLog.open('unmerge_identities', ctx)
+
+    try:
+        identities = _find_identities(uuids)
+    except NotFoundError as exc:
+        # At least one identity was not found, so they cannot be unmerged
+        raise exc
+
+    new_uids = []
```

Comment:
It will be better to rename this to `new_uidentities`. It's more clear because `uids` looks like you are storing identifiers instead of identities.

Reply:
Done.